### PR TITLE
Use torch 2.2 for daily CI (model tests)

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

`torch 2.2` is released 3 weeks ago, and our CircleCI already use it, not not daily CI so far.

This PR update the docker file to use torch 2.2 for daily CI, but only for the modeling tests.

(p.s. We are in the process of moving other tests to separate workflow files)

### CODE REVIEW

File: docker/transformers-all-latest-gpu/Dockerfile [hunk 1]
```diff
@@ -46,24 +50,6 @@ RUN python3 -m pip install --no-cache-dir git+https://github.com/huggingface/acc
...
-
-# Add aqlm for quantization testing
-RUN python3 -m pip install --no-cache-dir aqlm[gpu]==1.0.1
-
-# Add autoawq for quantization testing
-RUN python3 -m pip install --no-cache-dir https://github.com/casper-hansen/AutoAWQ/releases/download/v0.1.8/autoawq-0.1.8+cu118-cp38-cp38-linux_x86_64.whl
-
-# For bettertransformer + gptq
-RUN python3 -m pip install --no-cache-dir git+https://github.com/huggingface/optimum@main#egg=optimum
-
```
Comment (ArthurZucker) [inline docker/transformers-all-latest-gpu/Dockerfile:66]: this is related to the cleanup by marc right? 🤗

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks

File: docker/transformers-all-latest-gpu/Dockerfile (hunk 1 above)
Comment (ydshieh) [reply docker/transformers-all-latest-gpu/Dockerfile:66]: yes. This should be kept

> RUN python3 -m pip install --no-cache-dir git+https://github.com/huggingface/optimum@main#egg=optimum

however

Comment (ydshieh) [reply docker/transformers-all-latest-gpu/Dockerfile:66]: after talking to @younesbelkada

**Pull request merged**